### PR TITLE
Removing and creating field based on Permission

### LINKED ISSUE #190: Delete node when it no longer cannot be accessed after modification event
I'm close to ready for permission aware websocket notifications.
 
It will bring use case where public content gets turned into private. That will result a "modified"-event, but updating the node will eventually fail for 401 or 403 because gatsby develop has no longer permissions to read that page.

This should result in deletion of the node.

### PR DESCRIPTION
**Pull request by iFlameing**
Fixes #190 
Just one problem with a nested folder structure we are not providing URL of the nested child which is present in the Folder thus we are not able to delete node for them. can you help me with this How I handle this situation :)

### CODE REVIEW

**Comment by datakurre** [comment]
Could you rephrase? I didn’t fully understand the issue yet.

**Comment by datakurre** [comment]
Could you open an issue that we need to document ”live updates” for gatsby develop working best when ”all content types have publication workflow”.

You probably meant the case with default Plone where images and files don’t have workflow and their publication depends on their parents. Unfortunately that can be condigured for any types and there is no trivial solution to handle this.

**Comment by datakurre** [comment]
That said, somewhere there is a varoable of all known nodes that is saved as a state of the plugin that subsequent gatsby builds know to remove disappeared nodes. Maybe that could be used together with search.

Probably we should also learn to update the cache at some point.

I admit, I don’t currently remember the details of that ”state”, but might look into that later.

**Comment by iFlameing** [comment]
@datakurre let us consider a Folder Structure as ( outerfolder -> innerFolder -> newsItem). All of them is public. When we change the outer folder to Private and left inner folder and newsItem public. we get a modified event with child and parent. We will able to delete node of outerFolder but we can't delete node for innerFolder and newsItem. How I can delete these nodes?

**Comment by datakurre** [comment]
I still don’t understand. I’m sorry. If the child remains public why it should be deleted.

Please, try to describe one more time :)

**Comment by iFlameing** [comment]
@datakurre  because we deleted the navigation and breadcrumbs node of outerfolder and we cannot navigate to the inner folder and newsitem. But you are right we don't need to delete the node of those inner folder and newsitem. I am thinking too much. You  can merge this now :)
